doc/source/train/user-guides/using-gpus.rst

```diff
@@ -167,3 +225,103 @@ resources to use 0 CPUs:
             "CPU": 0,
         }
     )
+
+
+.. _train-resource-guide:
+
+How many nodes, workers, and resources should I use?
```

Comment:
This is good information but I feel like it's really scattered right now and as a user it would be very hard to figure out what to concretely do. It's generally not clear to me if this is intended to be more catered to people who are starting distributed training, or people who already have a job and are facing concrete bottlenecks (maybe we want to apply this to both categories).

Sorry if this comment isn't super concrete, maybe we can separate this from the PR for now and think more how to architect this information.